### PR TITLE
Fixes #21821: Refactor without trait

### PR DESCRIPTION
**Pull request by VinceMacBuche**
https://issues.rudder.io/issues/21821

### CODE REVIEW

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/CampaignApi.scala [hunk 1]
```diff
@@ -165,7 +165,7 @@ class CampaignApi (
             case eb: EmptyBox => Unexpected((eb ?~! "error when accessing request body").messageChain).fail
             case Full(bytes)  => campaignSerializer.parse(new String(bytes, charset))
           }
-        withId = if (campaign.info.id.value.isEmpty) campaign.copyWithId(CampaignId(stringUuidGenerator.newUuid)) else campaign
+        withId = if (campaign.info.id.value.isEmpty) campaign.copy(info = campaign.info.copy(id = CampaignId(stringUuidGenerator.newUuid))) else campaign
```
Comment (fanf) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/CampaignApi.scala:168]: you will want quicklens for that in a further update

**Comment by VinceMacBuche** [comment]
PR rebased

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by VinceMacBuche** [comment]
Commit modified

**Comment by fanf** [comment]
Things diverged too much, and the draft seems abandoned, so I'm closing it

**Pull request closed without merging**